### DISCUSSION TITLE
setting CatBoost to CPU whilst having default classifiers to utilize GPU when available.

### DISCUSSION THREAD

**Discussion by sgapple** [autogluon/autogluon#3197] [Q&A] [answered]
I'm current running: predictor = TabularPredictor(label=label, path=None).fit(train_data, ag_args_fit={'num_gpus': 0}), however, i find that CatBoost will be unstable causing the kernel to die  whenever the dataset gets too big while running with GPU. Therefore, is it possible to set CatBoost to CPU, whilst having the remaining classifiers(default) utilize GPU when available. 

In addition, I  have also experienced overheating causing the machine to hang when running CatBoost with CPU. However, training will usually passed after a force reboot. Is this normal?

ps. the data frame has about 15,000 rows, 200 to 600 columns; machine is running on Ubuntu with i9-13900 and rtx 3090TI.
Kind advise

**Comment by gradientsky** [comment, marked as answer]
You can set gpu control on model level - see [docs](https://auto.gluon.ai/stable/tutorials/tabular/advanced/tabular-gpu.html).

When training CatBoost on GPU, please consider setting [`gpu_ram_part`](https://catboost.ai/en/docs/references/training-parameters/performance#gpu_ram_part) to lower values: i.e. `0.8` (default is `0.95`). I found this reduced likelihood of it running out of memory.

**Reply by sgapple** [reply]
gradientsky, thank you for the reply. 
i pulled the default(get_hyperparameter_config) and added 'gpu_ram_part': 0.8 (tried 0.5), however,  it still crashed. Here is the snip:

  hyperparameters = {'NN_TORCH': {},
 'GBM': [{'extra_trees': True, 'ag_args': {'name_suffix': 'XT'}},
  {},
  'GBMLarge'],
 'CAT': {'gpu_ram_part': 0.8},
 'XGB': {},
 'FASTAI': {},
 'RF': [{'criterion': 'gini',
   'ag_args': {'name_suffix': 'Gini',
    'problem_types': ['binary', 'multiclass']}},
  {'criterion': 'entropy',
   'ag_args': {'name_suffix': 'Entr',
    'problem_types': ['binary', 'multiclass']}},
  {'criterion': 'squared_error',
   'ag_args': {'name_suffix': 'MSE',
    'problem_types': ['regression', 'quantile']}}],
 'XT': [{'criterion': 'gini',
   'ag_args': {'name_suffix': 'Gini',
    'problem_types': ['binary', 'multiclass']}},
  {'criterion': 'entropy',
   'ag_args': {'name_suffix': 'Entr',
    'problem_types': ['binary', 'multiclass']}},
  {'criterion': 'squared_error',
   'ag_args': {'name_suffix': 'MSE',
    'problem_types': ['regression', 'quantile']}}],
 'KNN': [{'weights': 'uniform', 'ag_args': {'name_suffix': 'Unif'}},
  {'weights': 'distance', 'ag_args': {'name_suffix': 'Dist'}}]}
  
  predictor = TabularPredictor(label=label, path=None).fit(train_data, verbosity=0, ag_args_fit={'num_gpus': 1}, hyperparameters=hyperparameters)
